Automattic/jetpack
# CRM: Profile summary - removing deleted invoice counts and totals

## Proposed changes:

* This PR ensures that total amounts (current total value and invoice total, plus the invoice count in brackets) are factoring in deleted invoices (by that I mean not including the deleted invoices), on contact and company profile summaries. It also corrects the invoice total counts and amount on both the Contact and Company 'View All' pages as well.

### Other information:


## Jetpack product discussion

https://github.com/Automattic/zero-bs-crm/issues/2673

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Testing locally on trunk, or on a test site with the latest stable version of Jetpack CRM, create both a company and contact with multiple invoices attached to them. On the profile summary, view the current total value and invoice total (plus count in brackets next to the total).
  * Contacts: `/wp-admin/admin.php?page=zbs-add-edit&action=view&zbstype=contact&zbsid={contactid}
  * Companies: `http://localhost/wp-admin/admin.php?page=zbs-add-edit&action=view&zbstype=company&zbsid={companyid}`
* Take note of the totals and invoice count - these should include the totals from deleted invoices.
* You can also test the totals box on the contact 'View All' page - the invoice total (and total) should exclude deleted invoices as well: `/wp-admin/admin.php?page=manage-customers`.
* In the developer console on that same contacts 'View All' page, type in `jpcrm_totals_table` and hit enter. You should see the updated totals, including some new variables such as `invoices_total_inc_deleted` and `total_sum_inc_deleted` that can be used at any stage.
* Another thing to test is the modified list view for both contacts and companies on their respective 'View All' pages, adding in invoice counts and invoice totals: https://kb.jetpackcrm.com/knowledge-base/how-do-i-change-columns-on-contact-list-view/
* These should also be correct (not including deleted invoices)
* Using the [Jetpack Beta plugin](https://github.com/Automattic/jetpack-beta) on a test site with this branch applied or testing locally with the branch checked out, open up the same pages again. The counts should be reduced to factor in deleted invoices.

## Screenshots:

A contact profile summary before a $3 deleted invoice was removed:
<img width="812" alt="Screenshot 2023-04-19 at 09 34 29" src="https://user-images.githubusercontent.com/16754605/233017792-868a41e7-e5f7-4c5a-93a3-bdbb844b230f.png">


And after:

<img width="798" alt="Screenshot 2023-04-19 at 09 33 16" src="https://user-images.githubusercontent.com/16754605/233017804-08657f5b-6de5-477e-9c89-5f8f30039003.png">

Info box changes:

<img width="299" alt="Screenshot 2023-04-19 at 09 40 45" src="https://user-images.githubusercontent.com/16754605/233019397-fb693265-c1f2-48cb-b9f5-a8612e3703f0.png">
<img width="275" alt="Screenshot 2023-04-19 at 09 40 11" src="https://user-images.githubusercontent.com/16754605/233019416-13d517f5-add1-4cea-862d-8c1d8b22abad.png">

Contacts 'View All' page including invoice totals and counts in the list, plus the totals box (bottom right):

<img width="1747" alt="Screenshot 2023-05-05 at 10 50 10" src="https://user-images.githubusercontent.com/16754605/236427286-b81c3443-b598-46b3-9b06-84b30158a38f.png">
